middle_end/flambda/from_lambda/cps_conversion.ml

```diff
@@ -325,76 +523,119 @@ let rec_catch_for_for_loop
                 Lstaticraise (cont, [next_value_of_counter]),
                 L.lambda_unit))),
           L.lambda_unit)))
-  in lam
+  in env, lam
 
-let rec cps_non_tail env (lam : L.lambda) (k : Ident.t -> Ilambda.t)
+let let_cont_with_extra_params env ~continuation_on_stack ~is_exn_handler
+      ~params ~(recursive : Asttypes.rec_flag)
```

Comment:
Also, this function doesn't seem to be used for recursive handlers. Maybe you could remove the `recursive` parameter and reflect that in the function's name ?